client.go

```diff
@@ -185,9 +185,13 @@ func (c *Client) exchange(m *Msg, co *Conn) (r *Msg, rtt time.Duration, err erro
 	}
 
 	co.SetReadDeadline(time.Now().Add(c.getTimeoutForRequest(c.readTimeout())))
-	r, err = co.ReadMsg()
-	if err == nil && r.Id != m.Id {
-		err = ErrId
+	for {
```

Comment:
Let’s guard this in a `if _, ok := co.Conn.(net.PacketConn); ok {` and keep the old behaviour for a TCP connection. Then it needs appropriate tests as well unfortunately.